### PR TITLE
 AP_Math: move location_passed_point and line_path_proportion to Location

### PR DESCRIPTION
**Pull request by khancyr**
location_passed_point was rename to passe_line_point, any other suggestion ?

### CODE REVIEW

**Comment by khancyr** [comment]
@peterbarker done !

File: libraries/AP_Common/Location.h [hunk 1]
```diff
@@ -108,6 +108,20 @@ class Location
     // return true when lat and lng are within range
     bool check_latlng() const;
 
+    // see if location is past a line parallel to
```
Comment (peterbarker) [inline libraries/AP_Common/Location.h]: You appear to have changed this comment to be incorrect.

File: libraries/AP_Common/Location.h [hunk 2]
```diff
@@ -108,6 +108,20 @@ class Location
...
+    // the line between point1 and point2. If point1 is
+    // our previous waypoint and point2 is our target waypoint
+    // then this function returns true if we have flown past
+    // the target waypoint
+    bool past_line_point(const Location &point1, const Location &point2) const;
```
Comment (peterbarker) [inline libraries/AP_Common/Location.h]: Not happy with this name.

Perhaps `past interval_finish_line`?

File: libraries/AP_Common/Location.h [hunk 3]
```diff
@@ -108,6 +108,20 @@ class Location
     // return true when lat and lng are within range
     bool check_latlng() const;
 
+    // see if location is past a line parallel to
+    // the line between point1 and point2. If point1 is
+    // our previous waypoint and point2 is our target waypoint
+    // then this function returns true if we have flown past
```
Comment (peterbarker) [inline libraries/AP_Common/Location.h]: s/flown/travelled/

Comment (peterbarker) [reply libraries/AP_Common/Location.h]: // see if location is past a line perpendicular to
// the line between point1 and point2 and passing through point2.
// If point1 is
// our previous waypoint and point2 is our target waypoint
// then this function returns true if we have flown past
// the target waypoint

File: libraries/AP_Scripting/lua_bindings.cpp [hunk 4]
```diff
@@ -244,9 +244,8 @@ static int location_passed_point(lua_State *L) {
         return luaL_argerror(L, args, "too many arguments");
     }
 
-    lua_pushboolean(L, location_passed_point(*check_location(L, -3),
-                                             *check_location(L, -2),
-                                             *check_location(L, -1)));
+    lua_pushboolean(L, (*check_location(L, -3)).past_line_point(*check_location(L, -2),
+                                                                   *check_location(L, -1)));
```
Comment (peterbarker) [inline libraries/AP_Scripting/lua_bindings.cpp]: whitespace

File: libraries/AP_Scripting/lua_bindings.cpp [hunk 5]
```diff
@@ -257,9 +256,8 @@ static int location_path_proportion(lua_State *L) {
         return luaL_argerror(L, args, "too many arguments");
     }
 
-    lua_pushnumber(L, location_path_proportion(*check_location(L, -3),
-                                               *check_location(L, -2),
-                                               *check_location(L, -1)));
+    lua_pushnumber(L, (*check_location(L, -3)).line_path_proportion(*check_location(L, -2),
+                                                                     *check_location(L, -1)));
```
Comment (peterbarker) [inline libraries/AP_Scripting/lua_bindings.cpp]: whitespace

**Comment by khancyr** [comment]
I have compile on scripting ... but not test as I don't know how it works

**Comment by WickedShell** [comment]
@khancyr I pushed in a large change to scripting so you will need to rebase to fix the changes. The good news though is that if you look in master basically all the code you were trying to edit disappeared form that file, and moved into a generated file, which doesn't use passed point at all :)

**Comment by khancyr** [comment]
correct rebase

File: libraries/AP_Math/examples/location/location.cpp [hunk 6]
```diff
@@ -55,7 +55,7 @@ static void test_passed_waypoint(void)
         struct Location loc = location_from_point(test_points[i].location);
         struct Location wp1 = location_from_point(test_points[i].wp1);
         struct Location wp2 = location_from_point(test_points[i].wp2);
-        if (location_passed_point(loc, wp1, wp2) != test_points[i].passed) {
+        if (loc.passed_line_point(wp1, wp2) != test_points[i].passed) {
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply libraries/AP_Math/examples/location/location.cpp]: yes, it should be past_point or past_line_point I think.

**Comment by tridge** [comment]
looks like its all merged now, closing

**Comment by itayguy** [comment]
line_path_proportion - can someone have a look on the math , It looks wrong to me !

**Pull request closed without merging**